ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -315,10 +315,14 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
         return false;
       }
 
-      if (!ci.type.equals(CompactionType.REBALANCE) && ci.numberOfBuckets > 0) {
-        if (LOG.isWarnEnabled()) {
-          LOG.warn("Only the REBALANCE compaction accepts the number of buckets clause (CLUSTERED INTO {N} BUCKETS). " +
-              "Since the compaction request is {}, it will be ignored.", ci.type);
+      boolean insertOnly = AcidUtils.isInsertOnlyTable(table.getParameters());
+      if (LOG.isWarnEnabled() && ci.type.equals(CompactionType.REBALANCE) && insertOnly) {
+        LOG.warn("REBALANCE compaction requested on an insert-only table ({}). Falling back to MAJOR compaction as " +
+            "REBALANCE compaction is supported only on full-acid tables", table.getTableName());
+        if (ci.numberOfBuckets > 0) {
+          LOG.warn("Only REBALANCE compaction on a full-acid table accepts the number of buckets clause " +
+              "(CLUSTERED INTO {N} BUCKETS). Since the compaction request is {} and the table is {}, it will be ignored.",
```

Comment:
why do we need a table type param here as it's only logged when table is insert only? that is why i was confused with the above if change

Reply:
Makes sense, I think originally I created this log entry elsewhere, and I forgot to adjust the message when it got moved inside the if.